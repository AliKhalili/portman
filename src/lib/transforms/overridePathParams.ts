/* eslint-disable @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any */
import pluralize from 'pluralize'

export const overridePathParams = (obj: any): any => {
  obj.item.map(resource => {
    const singular = pluralize.singular(resource.name.toLowerCase())
    const variableName = `{{${singular}Id}}`
    resource.item
      .filter(({ request: { url } }) => {
        return url.path.includes(':id')
      })
      .map(item => {
        item.request.url.variable.map(item => {
          if (item.key === 'id') {
            item.value = variableName
          }
        })
      })
  })
  return obj
}
