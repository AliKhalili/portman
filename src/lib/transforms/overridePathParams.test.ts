import { overridePathParams } from './overridePathParams'

describe('overridePathParams()', () => {
  it('should replace value of path param with a {{namedId}} variable', () => {
    const obj = {
      item: [
        {
          name: 'Monkeys',
          item: [
            {
              request: {
                url: {
                  path: ['crm', 'monkies', ':id'],
                  variable: [
                    {
                      disabled: false,
                      type: 'any',
                      value: '<string>',
                      key: 'id',
                      description: '(Required) ID of the monkey you are acting upon.'
                    }
                  ]
                }
              }
            }
          ]
        }
      ]
    }

    const transform = overridePathParams(obj)
    expect(transform).toMatchSnapshot()
  })
})
