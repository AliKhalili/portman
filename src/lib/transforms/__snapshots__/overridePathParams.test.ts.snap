// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`overridePathParams() should replace value of path param with a {{namedId}} variable 1`] = `
Object {
  "item": Array [
    Object {
      "item": Array [
        Object {
          "request": Object {
            "url": Object {
              "path": Array [
                "crm",
                "monkies",
                ":id",
              ],
              "variable": Array [
                Object {
                  "description": "(Required) ID of the monkey you are acting upon.",
                  "disabled": false,
                  "key": "id",
                  "type": "any",
                  "value": "{{monkeyId}}",
                },
              ],
            },
          },
        },
      ],
      "name": "Monkeys",
    },
  ],
}
`;
